challenges/08-coding-interview-prep/rosetta-code.json

```diff
@@ -12434,6 +12434,1022 @@
+        {
+          "text":
+            "'<code>derangements(\"567\")</code> should return <code>'+JSON.stringify(results[0])+'</code>.'",
+          "testString":
+            "assert.deepEqual(derangements(\"567\"),results[0],'<code>derangements(\"567\")</code> should return <code>'+JSON.stringify(results[0])+'</code>.');"
+        },
+        {
+          "text":
+            "'<code>derangements(\"xyz\")</code> should return <code>'+JSON.stringify(results[1])+'</code>.'",
+          "testString":
+            "assert.deepEqual(derangements(\"xyz\"),results[1],'<code>derangements(\"xyz\")</code> should return <code>'+JSON.stringify(results[1])+'</code>.');"
+        },
+        {
+          "text":
+            "'<code>derangements(\"mqr\")</code> should return <code>'+JSON.stringify(results[2])+'</code>.'",
+          "testString":
+            "assert.deepEqual(derangements(\"mqr\"),results[2],'<code>derangements(\"mqr\")</code> should return <code>'+JSON.stringify(results[2])+'</code>.');"
+        },
+        {
+          "text":
+            "'<code>derangements(\"1234\")</code> should return <code>'+JSON.stringify(results[3])+'</code>.'",
+          "testString":
+            "assert.deepEqual(derangements(\"1234\"),results[3],'<code>derangements(\"1234\")</code> should return <code>'+JSON.stringify(results[3])+'</code>.');"
+        },
+        {
+          "text":
+            "'<code>derangements(\"abcd\")</code> should return <code>'+JSON.stringify(results[4])+'</code>.'",
+          "testString":
+            "assert.deepEqual(derangements(\"abcd\"),results[4],'<code>derangements(\"abcd\")</code> should return <code>'+JSON.stringify(results[4])+'</code>.');"
+        }
+      ],
+      "id": "5a23c84252665b21eecc7f6d",
+      "challengeType": 5,
+      "releasedOn": "August 15, 2018",
+      "files": {
+        "indexjs": {
+          "key": "indexjs",
+          "ext": "js",
+          "name": "index",
+          "contents": ["function derangements (str) {", "  // Good luck!", "}"],
+          "head": [],
+          "tail": [
+            "var results=[",
+            "[\"675\",\"756\"],",
+            "[\"yzx\",\"zxy\"],",
+            "[\"qrm\",\"rmq\"],",
+            "[\"2143\",\"2341\",\"2413\",\"3142\",\"3412\",\"3421\",\"4123\",\"4312\",\"4321\"],",
+            "[\"badc\",\"bcda\",\"bdac\",\"cadb\",\"cdab\",\"cdba\",\"dabc\",\"dcab\",\"dcba\"]]"
+          ]
+        }
+      }
+    },
+    {
+      "title": "Permutations/Rank of a permutation",
+      "description": [
+        "A particular ranking of a permutation associates an integer with a particular ordering of all the permutations of a set of distinct items.",
+        "For our purposes the ranking will assign integers $0 .. (n! - 1)$ to an ordering of all the permutations of the integers $0 .. (n - 1)$.",
+        "For example, the permutations of the digits zero to 3 arranged lexicographically have the following rank:",
+        "<pre>  PERMUTATION      RANK",
```

Comment:
Unfortunately this will have to be as a single line. Should look something like:

```
"<pre>  PERMUTATION      RANK\n  (0, 1, 2, 3) ->  0\n  (0, 1, 3, 2) ->  1\n  (0, 2, 1, 3) ->  2\n  (0, 2, 3, 1) ->  3\n  (0, 3, 1, 2) ->  4...</pre>"
```